src/backend/pipeline/cont_scheduler.c

```diff
@@ -409,6 +423,10 @@ cq_bgproc_main(Datum arg)
 		break;
 	case Worker:
 		am_cont_worker = true;
+
+		/* XXX - put this somewhere nicer */
+		AdhocMgrDeleteAdhocs();
```

Comment:
Why is it here? Can't we put this in the start up of `cont_scheduler`? Or does this require a connection to a database?